api/v1alpha1/reference_types.go

```diff
@@ -24,7 +24,6 @@ type CrossNamespaceObjectReference struct {
 	APIVersion string `json:"apiVersion,omitempty"`
 
 	// Kind of the referent
-	// +kubebuilder:validation:Enum=GitRepository
```

Comment:
This has removed input verification for SourceRef.

Reply:
Please keep that struct for the source ref but rename it to `CrossNamespaceSourceReference`.